src/components/PeoplePage/PeoplePage.tsx

```diff
@@ -0,0 +1,62 @@
+import React, { useState, useEffect } from 'react';
+import { Person } from '../../types';
+import { Loader } from '../Loader';
+import { getPeople } from '../../api';
+import { PeopleTable } from '../PeopleTable/PeopleTable';
+
+export const PeoplePage: React.FC = () => {
+  const [people, setPeople] = useState<Person[]>([]);
+  const [isTableLoading, setIsTableLoading] = useState(false);
+  const [error, setError] = useState('');
+
+  const isTableVisible = !isTableLoading && !error;
+  const isErrorVisible = !isTableLoading && error;
+
+  useEffect(() => {
+    setIsTableLoading(true);
+
+    getPeople()
+      .then(visiblePeople => {
+        const peopleWithParents = visiblePeople.map(
+          (person, _, serverPeople) => {
+            const mother = serverPeople.find(m => m.name === person.motherName);
+            const father = serverPeople.find(f => f.name === person.fatherName);
```

Comment:
The right decision, but now you have two searches in the middle of another search, which is algorithmically not optimal.
To improve the algorithm, you can create a separate map (`peopleMap`) and add all the people to it using their name as a key (`peopleMap.set(person.name, person)`), after that go through the `people` array and find the person's father and mother (`peopleMap.get(person.motherName)`, `peopleMap.get(person.fatherName)`).
This is a more complicated approach, but more correct on real projects.